src/Parsers/examples/YAML_fuzzer.cpp

```diff
@@ -0,0 +1,41 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <time.h>
+#include <filesystem>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t * data, size_t size)
+{
+    /// How to test:
+    /// build ClickHouse with YAML_fuzzer.cpp
+    /// ./YAML_fuzzer YAML_CORPUS
+    /// where YAML_CORPUS is a directory with different YAML configs for libfuzzer
+
+    srand(time(NULL));
+    std::string cur_file = std::to_string(rand());
+
+    while (std::filesystem::exists(cur_file))
+    {
+        std::string cur_file = std::to_string(rand());
+    }
+
+    std::string input = std::string(reinterpret_cast<const char*>(data), size);
+    DB::YAMLParser parser;
+
+    std::ofstream temp_file(cur_file);
+    temp_file << input;
+    temp_file.close();
+
+    try
+    {
+        DB::YAMLParser::parse(cur_file);
+    }
+    catch (const DB::Exception&)
+    {
+        std::cout << "YAMLParser exception from bad file, etc. OK\n";
+    }
+
+    remove(cur_file.c_str());
```

Comment:
If you only wanted to create a temporary file then it's better to prefer one of the standard solutions - see `std::tmpnam()` or `Poco::TemporaryFile::tempName()`.